psu-img: Update repository firmware image directory

faisal-awada opened the pull request:
Removed firmware and MANIFEST files from device-images directory and added new directory format to add subdirectory with the PSU CCIN. The new format structure to have a CCIN directory and in the new directory shall have the MANIFEST and firmware image.

Added new 51DA and 51E9 directory along with FWR and updated MANIFEST files.

Change-Id: Ia5db24225f1809d94c8995103c28d405a1f5502a

faisal-awada committed:
psu-img: Update repository firmware image directory

Removed firmware and MANIFEST files from device-images directory and
added new directory format to add subdirectory with the PSU CCIN. The
new format structure to have a CCIN directory and in the new directory
shall have the MANIFEST and firmware image.

Added new 51DA and 51E9 directory along with FWR and updated MANIFEST
files.

Change-Id: Ia5db24225f1809d94c8995103c28d405a1f5502a
Signed-off-by: Faisal Awada <faisal@us.ibm.com>

derekhoward55 commented on 51DA/MANIFEST:
```diff
@@ -0,0 +1,3 @@
+purpose=xyz.openbmc_project.Software.Version.VersionPurpose.PSU
+version=393141313900
```
how did you determine this version number?  looks like this corresponds to "19.1A.09", but I don't think the 00 is valid.  Was this at offset 0x10 in the image?

faisal-awada replied:
The FW level pulled at offset 0x36 for AEI, then converted to ascii and reversed bytes.

derekhoward55 commented on 51E9/MANIFEST:
```diff
@@ -0,0 +1,3 @@
+purpose=xyz.openbmc_project.Software.Version.VersionPurpose.PSU
+version=393141313900
```
same question as above?

faisal-awada replied:
The latest FW IBM_RANIER_BOOTLOAD_2300W_1A62C_2E.2F.2D.hex will show version=453246324432

derekhoward55 commented on README.md:
```diff
@@ -1 +1,22 @@
\ No newline at end of file
+OpenBMC power supply firmware
+
+This directory contains P11 power supply firmware images supplied by the
```
I would remove the "P11" text, so it doesn't need to be modified later in P12

derekhoward55 commented on README.md:
```diff
@@ -1 +1,22 @@
+
+This directory contains P11 power supply firmware images supplied by the
+vendors. Each PSU firmware should reside in a directory named as the PSU CCIN.
+The firmware image file prefix should be replaced with IBM PSU CCIN and postfix
```
I would remove the "IBM" text, so other companies can use the same naming convention.

faisal-awada replied:
I will remove IBM, should I replace CCIN with model number?

derekhoward55 replied:
Not sure that it matters, whatever it's called in inventory manager is fine

derekhoward55 commented on README.md:
```diff
@@ -1 +1,22 @@
+    version=003000390031
+    extended_version=model=2B1E,manufacturer=
+
+Note: Files don't exist in this directory will require addition in
```
perhaps explain why

smccarney replied:
I'm not sure what this sentence is saying.  Do you mean "Note: Adding new files to this repository will require changes to ..." ?

derekhoward55 reviewed (commented):
See individual comments...

smccarney commented on README.md:
```diff
@@ -1 +1,22 @@
+
+This directory contains P11 power supply firmware images supplied by the
+vendors. Each PSU firmware should reside in a directory named as the PSU CCIN.
+The firmware image file prefix should be replaced with IBM PSU CCIN and postfix
```
Just curious, why do they need to rename their image file if it is already in a sub-directory that has the CCIN?

smccarney commented on README.md:
```diff
@@ -1 +1,22 @@
+This directory contains P11 power supply firmware images supplied by the
+vendors. Each PSU firmware should reside in a directory named as the PSU CCIN.
+The firmware image file prefix should be replaced with IBM PSU CCIN and postfix
+must remain as is. The PSU CCIN dir should contains the firmware file and
```
Nit: "should contain"

smccarney commented on README.md:
```diff
@@ -1 +1,22 @@
+must remain as is. The PSU CCIN dir should contains the firmware file and
+MANIFEST file. The contents of CCIN dir can be generated in 2 ways:
+
+* - Use tool to Generate Tarball with PSU image and MANIFEST Script, the
```
This is kind of a run-on sentence with some missing words.  Also, why is "Generate Tarball" capitalized?

Possible rewording:

Use the tool at phosphor-psu-code-mgmt/tools/generate-psu-tar to generated a tarball with the PSU image and MANIFEST file.  Then extract the tarball into the PSU CCIN directory.

smccarney commented on README.md:
```diff
@@ -1 +1,22 @@
+    the tarball into the PSU CCIN dir.
+OR 
+
+* - Add firmware file to PSU CCIN Dir and create file with the following
```
I assume they don't want the file to contain *exactly* those contents right?  I'd think the version and extended_version would vary?  Maybe say "create a MANIFEST file with contents similar to the following:" ?

faisal-awada replied:
It should say: Add firmware file to PSU CCIN Dir and create MANIFEST file with  contents similar to the following:

faisal-awada replied:
version and extended_version have same value if different phosphor-psu-code-manager will not proceed. I have asked the author about it (no reply).

faisal-awada committed:
psu-img: Remove previous firmware files

Removing previous firmware files from the repo.

Change-Id: I1b2ccadfa94c959c9404ffcb6bc7c13f37828d25
Signed-off-by: Faisal Awada <faisal@us.ibm.com>

faisal-awada committed:
Add PSU Firmware and resolve review issues

Changes made:
1 - Add new firmware for PSU models 51DA and 51E9. The newly committed
    firmware file, `IBM_RANIER_BOOTLOAD_2300W_1A62C_2E.2F.2D.hex` will
    display level 453246324432 on the system.
2 - Modified MANIFEST file for each PSU model subdir to reflect the new
    firmware verion.
3 - Resolved review issues

Change-Id: I5366e5814bd893ed019981a42e3121acf07ef920
Signed-off-by: Faisal Awada <faisal@us.ibm.com>

derekhoward55 reviewed (commented):
is the level number 2D.2F.2E?  AEI stores the bytes in reverse order of what the device driver will return them apparently.  The level in the manifest file should be compared exactly to what is in fw_version in sysfs.

smccarney reviewed (approved):
The README.md looks fine to me.  I don't know much about the directory structure and manifest files, so I will defer to Derek's review on those.

derekhoward55 commented:
Ok looks like the manifest files' fw_version is correct:

Enter the file name: IBM_RANIER_BOOTLOAD_2300W_1A62C_2E.2F.2D.hex
Enter version offset: 54
String read:  b'2D2F2E'
convert to ascii_tring and reversed bytes:  2D2F2E
FW version:  453246324432

I'd add the exact image file name to the commit message so we know which AEI file is checked in, other than that looks good to me.

derekhoward55 reviewed (approved):
Looks good!
